heron/scheduler-core/tests/java/com/twitter/heron/scheduler/RuntimeManagerMainTest.java

```diff
@@ -204,6 +209,35 @@ public void testManageTopologyFailCall() throws Exception {
   }
 
   @PrepareForTest(ReflectionUtils.class)
+  @Test(expected = UpdateDryRunResponse.class)
+  public void testManageTopologyDryRun() throws Exception {
+    config = mock(Config.class);
+    when(config.getStringValue(ConfigKeys.get("TOPOLOGY_NAME"))).thenReturn(TOPOLOGY_NAME);
+
+    RuntimeManagerMain runtimeManagerMain = spy(new RuntimeManagerMain(config, MOCK_COMMAND));
+    // Valid state manager class
+    Mockito.when(config.getStringValue(ConfigKeys.get("STATE_MANAGER_CLASS"))).
+        thenReturn(IStateManager.class.getName());
+    PowerMockito.mockStatic(ReflectionUtils.class);
+    PowerMockito.doReturn(Mockito.mock(IStateManager.class))
+        .when(ReflectionUtils.class, "newInstance", Mockito.eq(IStateManager.class.getName()));
+
+    // Legal request
+    UpdateDryRunResponse mockResponse = new UpdateDryRunResponse(
+        TopologyAPI.Topology.getDefaultInstance(), config, mock(PackingPlan.class),
+        mock(PackingPlan.class), new HashMap<String, Integer>());
+    doNothing().when(runtimeManagerMain)
+        .validateRuntimeManage(any(SchedulerStateManagerAdaptor.class), eq(TOPOLOGY_NAME));
+    // Successfully get ISchedulerClient
+    ISchedulerClient client = mock(ISchedulerClient.class);
+    doReturn(client).when(runtimeManagerMain).getSchedulerClient(any(Config.class));
+    // Happy path
+    doThrow(mockResponse).when(runtimeManagerMain).
```

Comment:
aren't you trying to assert that the code throws this response without mocking it's behavior to do so?